src/core/tools/generateImageTool.ts

```diff
@@ -177,13 +179,23 @@ export async function generateImageTool(
 			}
 
 			// Create a temporary OpenRouter handler with minimal options
-			const openRouterHandler = new OpenRouterHandler({} as any)
+			// kilocode_change start
+			const openRouterHandler = openRouterApiKey
+				? new OpenRouterHandler({})
+				: new KilocodeOpenrouterHandler({ kilocodeToken: kiloCodeApiKey })
```

Comment:
Do we want to prefer openRouter here? Or should we flip this and make kilocode the default if `kiloCodeApiKey` is set?

Reply:
just discussed this with @kevinvandijk, we will make it an explicit choice.

the preference for OpenRouter was because of the `:free` model option